python_ta/checkers/forbidden_python_syntax_checker.py

```diff
@@ -0,0 +1,72 @@
+"""A checker for reporting on the disallowed use of various Python syntax.
+"""
+from astroid import nodes
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import only_required_for_messages
+from pylint.lint import PyLinter
+
+
+class ForbiddenPythonSyntaxChecker(BaseChecker):
+    """A checker class to report on the disallowed use of various Python syntax.
+
+    Includes optional checking for the following syntax:
+      - break
+      - continue
+      - comprehension
+      - for
+      - while
+
+    By default, all Python syntax is allowed. Use options to disallow.
+    """
+
+    name = "forbidden_python_syntax"
+    msgs = {
+        "E9950": (
+            "You may not use %s syntax in your code.",
+            "forbidden-python-syntax",
+            "Used when %s syntax is found in your code.",
+        ),
+    }
+    options = (
+        (
+            "disallowed-python-syntax",
+            {
+                "default": (),
+                "type": "csv",
+                "metavar": "<disallowed-syntax>",
+                "help": "List of Python syntax that are not allowed to be used.",
+            },
+        ),
+    )
+
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    @only_required_for_messages("forbidden-python-syntax")
+    def visit_default(self, node: nodes.NodeNG) -> None:
+        """Visit a node in the AST."""
+        name = _pascal_case_to_lower(node.__class__.__name__)
```

Comment:
So the main change here is I want you to just use the name of the astroid node class (i.e., `name = node.__class__.__name__` to do the comparison. That simplifies the code a little bit too.

Reply:
Fixed!